### PR TITLE
review comments

### PR DESCRIPTION
**Pull request by rhaas80**
Never intended to be merged, only as a vehicle for comments.

### CODE REVIEW

File: library/BasicStuff/hdf5imple.cc [hunk 1]
```diff
@@ -6,6 +6,7 @@ namespace EOS_Toolkit {
 void read(const h5attr& a, bool& b)
 {
   int i;
+  // RH: where is read(hattr& a, int& i) defined?
```
Comment (wokast) [inline library/BasicStuff/hdf5imple.cc:9]: It is a template in include/hdf5simple.h

**Comment by rhaas80** [comment]
the HDF5 files in EOS will not be visible to the user right now (they are in the tarball but not copied anywhere where the user can find them).  Should they be? 

The Python script to create HDF5 EOS files should be provided in exe/sim/foo.py

**Comment by rhaas80** [comment]
There must be a documenation.tex file, even if it is no more than http://einsteintoolkit.org/thornguide/WVUThorns/Baikal/documentation.html

**Comment by wokast** [comment]
> the HDF5 files in EOS will not be visible to the user right now (they are in the tarball but not copied anywhere where the user can find them). Should they be?

No, those are used for the unit tests/benchmarks  and are not supposed as the start of a public EOS catalog. Such a catalog would go in a separate repository in any case.

> 
> The Python script to create HDF5 EOS files should be provided in exe/sim/foo.py

Does ET have any way to distribute Python stuff? Because this file needs some packages such as numpy, h5py. In any case, my aim is to make all Python bindings of the library and the Python EOS file creation module available through pip (and maybe conda) in the near future.  It is also likely that the file-creation API and the library Python bindings will be merged in one Python package. Once that happens, the former is no longer just a single Python module file.

**Comment by rhaas80** [comment]
we could provide them in exe/sim/foo.py where other utilities live. Them requiring h5py is unfortunate but I would rather provide the python script and say "if you want to use them, you will need to provide these python packages" then not provide them at all.

**Comment by rhaas80** [comment]
so the EOS file creation tools require some RePrimAnd Python module? When I looked at them (and only the EOS file creation tools are what I am concerned right now, not the Python bindings for RePrimAnd), they seemed to not require anything RePrimAnd specific.

**Comment by rhaas80** [comment]
One issue with the docs: it may be better to host the generated HTML pages for the documentation on GitHub and not AEI resources so that the pages do not go away if / when the AEI terminates you account if / when / after you leave there.

**Comment by wokast** [comment]
> so the EOS file creation tools require some RePrimAnd Python module? When I looked at them (and only the EOS file creation tools are what I am concerned right now, not the Python bindings for RePrimAnd), they seemed to not require anything RePrimAnd specific.

Indeed, at the moment the file creation Python API is just one module file that does not depend on RePrimAnd's Python bindings. However, I plan to merge the two into one RePrimAnd Python package that can be installed through PyPi. 

Note however that the file creation is not a Python script, but a Python module. So I wonder if exe/sim is the right place for that?

**Comment by wokast** [comment]
> One issue with the docs: it may be better to host the generated HTML pages for the documentation on GitHub and not AEI resources so that the pages do not go away if / when the AEI terminates you account if / when / after you leave there.

The AEI was the quickest solution. When I find some time to learn howto teach github pages to display sphinx-generated html, I might move it there. Also considered readthedocs, but found out they place advertisement (yuck).

**Comment by rhaas80** [comment]
The simplest way to host on github would be to create a branch gh-pages (or so) which is not connected to the actual code but instead only contains that html pages you would like to show. Then tell github to serve this branch using

https://github.com/wokast/RePrimAnd/settings/pages

it will then appear at:

https://wokast.github.io/RePrimAnd

Running sphinx and committing would be up to you then.

This is eg what we did for https://github.com/EinsteinToolkit/et2021uiuc/

It's not automated though. If you wanted automation you would have to set up a github actions pipeline similar eg to the one here (though you can also just put everything into a shell script and have the pipeline run the shell script after the checkout):

https://github.com/mojamil/einsteintoolkit/blob/master/.github/workflows/main.yml

which drives:

https://mojamil.github.io/einsteintoolkit

**Comment by rhaas80** [comment]
> > so the EOS file creation tools require some RePrimAnd Python module? When I looked at them (and only the EOS file creation tools are what I am concerned right now, not the Python bindings for RePrimAnd), they seemed to not require anything RePrimAnd specific.
> 
> Indeed, at the moment the file creation Python API is just one module file that does not depend on RePrimAnd's Python bindings. However, I plan to merge the two into one RePrimAnd Python package that can be installed through PyPi.
> 
> Note however that the file creation is not a Python script, but a Python module. So I wonder if exe/sim is the right place for that?

Uh, I was thinking of: https://github.com/wokast/RePrimAnd/blob/public/EOS/create/reprimand_eos_format.py which is indeed a module I guess. I would consider it useful to provide for ET users in the same manner that providing the coal executables in LORENE2 is useful even though strictly speaking the LORENE2 / RePrimAnd ExternalLibrary thorns are only readers for the produced files. Is there an example on how to use the file creation module?

**Comment by wokast** [comment]
> Uh, I was thinking of: https://github.com/wokast/RePrimAnd/blob/public/EOS/create/reprimand_eos_format.py which is indeed a module I guess. I would consider it useful to provide for ET users in the same manner that providing the coal executables in LORENE2 is useful even though strictly speaking the LORENE2 / RePrimAnd ExternalLibrary thorns are only readers for the produced files. Is there an example on how to use the file creation module?

There is no example, but using it just requires to call a single function for the desired EOS type from the module. Those functions are documented under EOS FIles / Creating EOS Files.

How about making the module file available in the thorn folder, e.g. `arrangements/ExternalLibraries/RePrimAnd/tools/reprimand_eos_format.py`

**Comment by wokast** [comment]
I'll give github pages a try. Still recall further complications regarding the need to deactivate Jekyll.

**Comment by rhaas80** [comment]
Thanks for giving the documentation a try.

Anything that is in "ET_interface/thon/RePrimAnd" is visible and checked out to arrangements/ExternalLibraries/RePrimAnd so putting it into "tools" would work, yes. Traditionally the directory is "src/utils" not "tools" (and also traditionally things in there are then copied [after having been compiled] to exe/sim/ but that is not done for every single utility eg CarpetIOHDF5's SubstituteDeprecatedParameters.pl which does not require compilation).

**Comment by rhaas80** [comment]
I would delete the branch in my repository and we can then kill this pull request which was a dummy anyway. Would that be ok for you, Wolfgang?

**Comment by wokast** [comment]
Yes, go ahead.

**Comment by rhaas80** [comment]
done

**Pull request closed without merging**